src/textual_serve/app_service.py

```diff
@@ -268,7 +268,23 @@ async def on_meta(self, data: bytes) -> None:
         Args:
             data: Encoded meta data.
         """
-        meta_data = json.loads(data)
+        meta_data: dict[str, object] = json.loads(data)
+        meta_type = meta_data["type"]
 
-        if meta_data["type"] == "exit":
+        if meta_type == "exit":
             await self.remote_close()
+        elif meta_type == "open_url":
+            payload = json.dumps(
```

Comment:
We should probably factor this out in to a help method at some point. I'm assuming this will grow more packet types.